lib/src/rules/cast_nullable_to_non_nullable.dart

```diff
@@ -80,6 +80,7 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (expressionType != null &&
         type != null &&
         expressionType is! DynamicType &&
```

Comment:
It would have been more logical, if we checked all for `expressionType` first, and then `type`.
From performance point of view, neither type will ever be `null`.

Reply:
Updated. Thanks!